From: Bernhard Walle <bwalle@suse.de>
Subject: [PATCH] Fix build on x86-64 with CONFIG_SMP unset

Fix following compiler errors:

fs/squashfs/inode.c: In function ‘squashfs_get_cached_block’:
fs/squashfs/inode.c:372: error: implicit declaration of function ‘set_current_state’
fs/squashfs/inode.c:372: error: ‘TASK_UNINTERRUPTIBLE’ undeclared (first use in this function)
fs/squashfs/inode.c:372: error: (Each undeclared identifier is reported only once
fs/squashfs/inode.c:372: error: for each function it appears in.)
fs/squashfs/inode.c:374: error: implicit declaration of function ‘schedule’
fs/squashfs/inode.c:375: error: ‘TASK_RUNNING’ undeclared (first use in this function)
fs/squashfs/inode.c:404: error: ‘TASK_INTERRUPTIBLE’ undeclared (first use in this function)
fs/squashfs/inode.c: In function ‘release_cached_fragment’:
fs/squashfs/inode.c:500: error: ‘TASK_UNINTERRUPTIBLE’ undeclared (first use in this function)
fs/squashfs/inode.c:500: error: ‘TASK_INTERRUPTIBLE’ undeclared (first use in this function)
fs/squashfs/inode.c: In function ‘get_cached_fragment’:
fs/squashfs/inode.c:531: error: ‘TASK_UNINTERRUPTIBLE’ undeclared (first use in this function)
fs/squashfs/inode.c:534: error: ‘TASK_RUNNING’ undeclared (first use in this function)


Signed-off-by: Bernhard Walle <bwalle@suse.de>

---
 fs/squashfs/inode.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/squashfs/inode.c
+++ b/fs/squashfs/inode.c
@@ -31,6 +31,7 @@
 #include <linux/vfs.h>
 #include <linux/vmalloc.h>
 #include <linux/smp_lock.h>
+#include <linux/sched.h>
 
 #include "squashfs.h"
 
